draft-rha-jose-hpke-encrypt.md

```diff
@@ -208,51 +219,47 @@ This example demonstrates the representaton of an encapsulated key as a JWK.
+- The "aad" parameter MUST take the Additional Authenticated Data encryption parameter defined in Step 14 of Section 5.1 of {{RFC7516}} as input. 
 
-This is a deviation from the rule in Section 4.1.2 of {{RFC7516}}. 
-
-This example demonstrates the use of an encapsulated key with JSON Web Encryption Integrated Encryption and Compact Serialization:
+The following example demonstrates the use of Integrated Encryption with Compact Serialization:
 
 ~~~
 eyJhbGciOiJIUEtFLUJhc2UtUDI1Ni1TSEEyNTYtQUVTMTI4R0NNIiwiZXBrIjp7Imt0eSI6IkVLIiwiZWsiOiJCQU9TeWV3M05JLUkwNEd2WU1MT3Y0cDBEVG5WMWZjWnBFVW10dGs0YkRTdDAtakxzY0FDN3h3MjdORTFHZ0VuMUgtM3ZXSFA5eW1BOHl4aFRmVDBkYjQifX0...afBw3T1hUNjci4qq3ZZ-9KxnttB0iCEO_GUqbIStqYqB5DgRDpyYSuvoH1mMA31qKPqB41ld5mSP34yUys6WJM7nstDJ1-4nqUdhRpgfkGTECA.
 ~~~
-{: #compact-example align="left" title="Integrated Encryption Encryption"}
+{: #compact-example align="left" title="Integrated Encryption with Compact Serialization"}
 
 ### Key Encryption
 
-In Key Encryption mode, HPKE is used to encrypt the	Content Encryption Key (CEK), and the resulting ciphertext is included in the JWE ciphertext. The (compressed) plaintext will be encrypted using the CEK as explained in Step 15 of Section 5.1 of {{RFC7516}}.
+This mode support more than one recipient, and JSON Serializations.
+
+HPKE is used to encrypt the	Content Encryption Key (CEK), and the resulting ciphertext is included in the JWE ciphertext. 
+The plaintext will be encrypted using the CEK as explained in Step 15 of Section 5.1 of {{RFC7516}}.
 
-In this mode, JWE JSON serialization MUST be used. The sender MUST place the 'epk' and 'alg' parameters in the per-recipient unprotected header to indicate the	use of HPKE. Optionally, the per-recipient unprotected header MAY contain the 'kid' parameter used to identify the static recipient public key used by the sender. The 'enc' (Encryption Algorithm) parameter MUST be present to identify the content encryption algorithm used to perform encryption on the plaintext to produce the ciphertext. The "enc" Header Parameter MUST occur only within the JWE Protected Header.	 		
+The sender MUST place the 'epk' and 'alg' parameters in the per-recipient unprotected header to indicate the use of HPKE.  
+The "enc" (Encryption Algorithm) Header Parameter MUST be present in the JWE Protected Header.	 		
 
 In Key Encryption mode: 
 - The JWE Encrypted Key MUST be the base64url encoded 'ct' value.
 - The JWE Initialization Vector MUST be produced as described in { Section 5.1 of RFC7516 }
 - The JWE Authentication Tag MUST be produced as described in { Section 5.1 of RFC7516 }
 
-In JWE JSON Serialization, the following mechanisms MUST be selected to provide protection against an attacker who manipulates the encryption algorithm in the 'enc' parameter in the protected header. The attack is discussed in {{?I-D.draft-ietf-lamps-cms-cek-hkdf-sha256}}:
```

Comment:
I removed this.

I am not confident we need to say anything on this topic, I think the use of `aad` in seal and open, combined with the use of the JWE Protected Header in JSON serialization is sufficient.